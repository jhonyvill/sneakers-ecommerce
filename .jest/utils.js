import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../src/Themes";

export function renderWithTheme(children) {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}

export const productDefault = {
  id: "id-1",
  brand: "brand mock",
  title: "title mock",
  price: 250,
  discount: 0.5,
  description: "description mock",
  pictures: [
    {
      src: "image-1.jpg",
      thumb: "thumbnail-1.jpg",
      alt: "product 1",
    },
    {
      src: "image-2.jpg",
      thumb: "thumbnail-2.jpg",
      alt: "product 2",
    },
    {
      src: "image-3.jpg",
      thumb: "thumbnail-3.jpg",
      alt: "product 3",
    },
    {
      src: "image-4.jpg",
      thumb: "thumbnail-4.jpg",
      alt: "product 4",
    },
    {
      src: "image-5.jpg",
      thumb: "thumbnail-5.jpg",
      alt: "product 5",
    },
  ],
};

export const itemDefault = {
  product: { ...productDefault },
  quantity: 1,
};
