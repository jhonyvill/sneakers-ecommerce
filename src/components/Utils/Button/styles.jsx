import styled from "styled-components";

export const StyledButton = styled.button`
  background-color: ${({ theme }) => theme.colors.primaryOrange};
  width: 100%;
  min-width: 8.5rem;
  padding: 1rem 0;
  color: ${({ theme }) => theme.colors.white};
  font-weight: 700;
  font-size: ${({ theme }) => theme.sizes.s};
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0 0.625rem;
  height: 3.5rem;

  &:hover {
    opacity: 0.7;
  }

  & svg > * {
    fill: ${({ theme }) => theme.colors.white};
    scale: 0.8;
  }

  &.shadowActive {
    box-shadow: 0 20px 25px 0 ${({ theme }) => theme.colors.paleOrange};
  }

  @media (max-width: ${({ theme }) => theme.screen.tablet}) {
    gap: 0;
    & svg {
      scale: 0.75;
    }
  }
`;
