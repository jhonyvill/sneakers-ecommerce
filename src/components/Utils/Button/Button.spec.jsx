import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderWithTheme } from "../../../../.jest/utils";
import Button from "./index";

function renderButton(clickFunction = jest.fn()){
  return renderWithTheme(<Button onClick={clickFunction}>Add to cart</Button>);
}

describe("<Button />", () => {
  it('should render button with text "Add to cart"', () => {
    renderButton();

    const button = screen.getByRole("button", { name: "Add to cart" });
    expect(button).toBeInTheDocument();
  });

  it("should call function on button click", async () => {
    const fn = jest.fn();
    renderButton(fn);

    const button = screen.getByRole("button", { name: "Add to cart" });

    await userEvent.click(button);

    expect.assertions(1);
    expect(fn).toHaveBeenCalledTimes(1);
  });
});

describe("<Button /> snapshots", () => {
  it("should match snapshot", () => {
    const { container } = renderButton();
    expect(container.firstChild).toMatchSnapshot();
  });
});
