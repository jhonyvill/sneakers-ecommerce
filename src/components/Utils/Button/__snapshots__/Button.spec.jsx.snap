// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshots should match snapshot 1`] = `
<button
  class="styles__StyledButton-sc-1csa66j-0 hmYbLP"
>
  Add to cart
</button>
`;
